app/src/main/java/io/homeassistant/companion/android/sensors/DevicePolicyManager.kt

```diff
@@ -0,0 +1,70 @@
+package io.homeassistant.companion.android.sensors
+
+import android.content.Context
+import android.content.Intent
+import io.homeassistant.companion.android.common.sensors.SensorManager
+import io.homeassistant.companion.android.common.R
+
+class DevicePolicyManager : SensorManager {
+    companion object {
+        private const val TAG = "DevicePolicyMgr"
+
+        val isWorkProfile = SensorManager.BasicSensor(
+            "is_work_profile",
+            "binary_sensor",
+            R.string.sensor_name_work_profile,
+            R.string.sensor_description_work_profile
+        )
+    }
+
+    override fun docsLink(): String {
+        return "https://companion.home-assistant.io/docs/core/sensors#work-profile-sensor"
+    }
+
+    private var isManagedProfileAvailable: Boolean = false
+
+    override val enabledByDefault: Boolean
+        get() = false
+    override val name: Int
+        get() = R.string.sensor_name_device_policy
+
+    override fun getAvailableSensors(context: Context): List<SensorManager.BasicSensor> {
+        return listOf(isWorkProfile)
+    }
+
+    override fun requiredPermissions(sensorId: String): Array<String> {
+        return emptyArray()
+    }
+
+    override fun requestSensorUpdate(context: Context) {
+        updateWorkProfile(context)
```

Comment:
I don't think the code will reach this anymore right?

Reply:
In the normal flow of things that is true, but to minimize changes I did not change the interface, so I think this function still needs to be provided.

Reply:
The function does need to be there but the update call here will always return `false` so its code doesnt have any operation to provide a valid state if its hit.  In cases like this you can just have a blank function with a comment.